pkg/lib-volume-populator/populator-machinery/controller.go

```diff
@@ -799,6 +805,27 @@ func (c *controller) syncPvc(ctx context.Context, key, pvcNamespace, pvcName str
 	return nil
 }
 
+func (c *controller) retryFailedPopulator(ctx context.Context, pvc *corev1.PersistentVolumeClaim, namespace, podName, counter string) (err error) {
+	pvc.Annotations[AnnPopulatorReCreations] = counter
```

Comment:
how about converting the counter to string here? it makes more sense for counter to be a number when it's passed here